plugins/user-settings-backend/src/service/router.ts

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { errorHandler } from '@backstage/backend-common';
+import { AuthenticationError, InputError } from '@backstage/errors';
+import {
+  getBearerTokenFromAuthorizationHeader,
+  IdentityClient,
+} from '@backstage/plugin-auth-node';
+import express, { Request } from 'express';
+import Router from 'express-promise-router';
+
+import { UserSettingsStore } from '../database';
+
+/**
+ * @public
+ */
+export interface RouterOptions<T> {
+  userSettingsStore: UserSettingsStore<T>;
+  identity: IdentityClient;
+}
+
+/**
+ * Create the user settings backend routes.
+ *
+ * @public
+ */
+export async function createRouter<T>(
+  options: RouterOptions<T>,
+): Promise<express.Router> {
+  const { userSettingsStore, identity } = options;
+  const router = Router();
+  router.use(express.json());
+
+  /**
+   * Helper method to extract the userEntityRef from the request.
+   */
+  const getUserEntityRef = async (req: Request): Promise<string> => {
+    const token = getBearerTokenFromAuthorizationHeader(
```

Comment:
I am working on a patch to extract the user from the request.. #12542

It might be nice to integrate this with it, as soon as it is ready.

Reply:
it's merged. :)

Reply:
Nice 👍 Great timing! 😄